quantus/metrics/robustness_metrics.py

```diff
@@ -937,7 +942,7 @@ def __call__(
                     targets=y,
                     **self.kwargs,
                 )
-                a_perturbed = utils.expand_attribution_channel(a_batch, x_batch_s)
+                a_perturbed = utils.expand_attribution_channel(a_perturbed, x_input)[0]
```

Comment:
why only accessing the first element?

Reply:
because the code below currently expects a_perturbed to not have a batch_axis, but expand_attribution_channel is running on batches. This is due to the current indexing implementation (infer_attribution_axes, expand_indices, assert_indexed_axes) being implemented with arrays w/o a batch axis in mind, in order to fit the current paradigm of looping over each input. 

These functions will need to be slightly altered for the planned batching-PR.

Reply:
I will add a comment above that line.